plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -1146,6 +1152,185 @@ public void testExecuteProcedureWithInvalidQuery()
+                .isFullyPushedDown();
+        assertThat(query(session, "SELECT * FROM (SELECT * FROM %s WHERE regionkey < 3 LIMIT 2)".formatted(tableName)))
+                .skipResultsCorrectnessCheckForPushdown()
+                .isFullyPushedDown();
+        assertThat(query(session, "SELECT * FROM (SELECT * FROM %s) WHERE regionkey < 3 LIMIT 2".formatted(tableName)))
+                .skipResultsCorrectnessCheckForPushdown()
+                .isFullyPushedDown();
+        assertThat(query(session, "SELECT * FROM (SELECT * FROM %s WHERE regionkey < 3) LIMIT 2".formatted(tableName)))
+                .skipResultsCorrectnessCheckForPushdown()
+                .isFullyPushedDown();
+
+        // With native query
+        assertNativeQueryWithLimitPushdown(session, tableName);
+        // With Empty projection
+        assertEmptyProjectionWithLimitPushdown(session, tableName);
+    }
+
+    private void assertNativeQueryWithLimitPushdown(Session session, String tableName)
+    {
+        assertThat(query(session, "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT * FROM %s')) LIMIT 0".formatted(tableName)))
+                .returnsEmptyResult();
+        assertThat(query(session, "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT * FROM %s')) LIMIT 3".formatted(tableName)))
+                .skipResultsCorrectnessCheckForPushdown()
+                .isFullyPushedDown();
+        assertThat(query(session, "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT * FROM %s')) LIMIT 10".formatted(tableName)))
+                .isFullyPushedDown();
+
+        assertThat(query(session, "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT * FROM %s')) WHERE regionkey < 3 LIMIT 2".formatted(tableName)))
+                .skipResultsCorrectnessCheckForPushdown()
+                .isFullyPushedDown();
+        assertThat(query(session, "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT * FROM %s')) WHERE regionkey > 5 LIMIT 2".formatted(tableName)))
+                .returnsEmptyResult();
+    }
+
+    private void assertEmptyProjectionWithLimitPushdown(Session session, String tableName)
+    {
+        PlanMatchPattern aggregationOverTableScan = node(AggregationNode.class, anyTree(node(TableScanNode.class)));
+
+        assertThat(query(session, "SELECT COUNT(*) FROM %s LIMIT 0".formatted(tableName)))
+                .returnsEmptyResult();
+        assertThat(query(session, "SELECT COUNT(*) FROM %s LIMIT 1".formatted(tableName)))
+                .matches("VALUES BIGINT '5'")
+                .isNotFullyPushedDown(aggregationOverTableScan);
+        assertThat(query(session, "SELECT COUNT(*) FROM %s LIMIT 10".formatted(tableName)))
+                .matches("VALUES BIGINT '5'")
+                .isNotFullyPushedDown(aggregationOverTableScan);
+
+        assertThat(query(session, "SELECT COUNT(*) FROM (SELECT * FROM %s LIMIT 0)".formatted(tableName)))
+                .matches("VALUES BIGINT '0'");
```

Comment:
Why checking pushdown is missing here?

Reply:
As Praveen mentioned, for LIMIT 0, we only have `Values` node.

`EXPLAIN ANALYZE SELECT * FROM tpch.region LIMIT 0`

```
MaterializedResult{rows=[[Trino version: testversion
Queued: 1.78ms, Analysis: 546.19ms, Planning: 6.26ms, Execution: 124.63ms
Fragment 1 [SINGLE]
    CPU: 79.00us, Scheduled: 82.42us, Blocked 0.00ns (Input: 0.00ns, Output: 0.00ns), Input: 0 rows (0B); per task: avg.: 0.00 std.dev.: 0.00, Output: 0 rows (0B)
    Peak Memory: 0B, Tasks count: 1; per task: max: 0B
    Output layout: [regionkey, name, comment]
    Output partitioning: SINGLE []
    Values[]
        Layout: [regionkey:bigint, name:varchar, comment:varchar]
        Estimates: {rows: 0 (0B), cpu: 0, memory: 0B, network: 0B}
        CPU: 0.00ns (?%), Scheduled: 0.00ns (?%), Blocked: 0.00ns (?%), Output: 0 rows (0B)
        Input avg.: 0.00 rows, Input std.dev.: ?%

]], types=[varchar], setSessionProperties={}, resetSessionProperties=[]}
```